src/dataset/mod.rs

```diff
@@ -17,11 +17,16 @@ pub use resource::{Resource, Type as ResourceType};
 pub struct Dataset {
     pub title: String,
     pub description: String,
+    pub region: Option<String>,
     pub license: License,
+    pub contact_names: Vec<String>,
```

Comment:
Wouldn't it be nicer to have something like `contacts: Vec<Contact>` here with

```rust
pub struct Contact {
   pub name: String,
   pub email: String,
}
```

Reply:
Actually I was thinking of this, but decided to do an initial commit with a plain and simple version. I wanted to have kind of a struct for an email that allows only valid email formats: is this possible?

Reply:
This is possible, e.g. `pub struct EMail(String);` which wraps a string with all methods constructing it validating the format. I started out this way with URL and I think it is not helpful for us: We should be lenient and try to fix up inconsistent, i.e. accept all strings and fix-up possible errors like `mailto:` prefixes.

In this case, we can certainly add the `Contact` type later on (but it will also change how the harvester looks like). I would certainly want to make that change as it prevents an internally inconsistent state from happening where there are a different number of names than e-mail addresses.

Reply:
But this is actually happening on Wasser-DE side: There are entries with no name but an email address (e.g. 'Organisation für die Umsetzung der Wasserrahmenrichtlinie in Schleswig-Holstein') so we will have to work that out. On the other hand I agree that the way I implemented it potentially an email is associated with the wrong name.

Reply:
Ok, so we need something more elaborate like

```rust
pub struct Contact {
  pub name: String,
  pub email: Vec<String>,
}
```

to have zero or one or many e-mail addresses associated to a contact. (I think we should really crate a separate `contact` module like for license and tags because I expect this to get more complicated due to sanitization and alternative modes of contact like phone numbers.)